ctapipe/core/tests/test_qualityquery.py

```diff
@@ -8,33 +8,33 @@
 from ctapipe.core.traits import List
 
 
-def test_selector():
+def test_selector(subarray_prod5_paranal):
     """test the functionality of an example Selector subclass"""
 
     class ExampleQualityQuery(QualityQuery):
         """Available variables: x"""
 
         quality_criteria = List(
-            default_value=[
-                ("high_enough", "x > 3"),
-                ("a_value_not_too_high", "x < 100"),
-                ("smallish", "x < sqrt(100)"),
+            [
+                ("high_enough", [("type", "*", "x > 3")]),
+                ("a_value_not_too_high", [("type", "*", "x < 100")]),
+                ("smallish", [("type", "*", "x < np.sqrt(100)")]),
             ],
         ).tag(config=True)
 
-    query = ExampleQualityQuery()
+    query = ExampleQualityQuery(subarray=subarray_prod5_paranal)
 
-    criteria1 = query(x=0)  # pass smallish
+    criteria1 = query(tel_id=1, x=0)  # pass smallish
```

Comment:
if you set the `key` to default value None, this test won't need to be adapted at all.